kura/org.eclipse.kura.core.comm/src/main/java/org/eclipse/kura/core/comm/CommConnectionImpl.java

```diff
@@ -121,110 +118,115 @@ public DataOutputStream openDataOutputStream() throws IOException {
+
+        this.outputStream.write(message, 0, message.length);
+        this.outputStream.flush();
     }
 
     @Override
     public synchronized byte[] sendCommand(byte[] command, int timeout) throws KuraException, IOException {
-        if (command != null) {
-            s_logger.debug("sendMessage() - {}", getBytesAsString(command));
+        if (command == null) {
+            throw new NullPointerException("Serial command must not be null");
+        }
 
-            if (this.m_outputStream == null) {
-                openOutputStream();
-            }
-            if (this.m_inputStream == null) {
-                openInputStream();
-            }
+        if (logger.isDebugEnabled()) {
+            logger.debug("sendMessage() - {}", getBytesAsString(command));
+        }
 
-            byte[] dataInBuffer = flushSerialBuffer();
-            if (dataInBuffer != null && dataInBuffer.length > 0) {
-                s_logger.warn("eating bytes in the serial buffer input stream before sending command: "
-                        + getBytesAsString(dataInBuffer));
-            }
-            this.m_outputStream.write(command, 0, command.length);
-            this.m_outputStream.flush();
-
-            ByteBuffer buffer = getResponse(timeout);
-            if (buffer != null) {
-                byte[] response = new byte[buffer.limit()];
-                buffer.get(response, 0, response.length);
-                return response;
-            } else {
-                return null;
-            }
+        if (this.outputStream == null) {
+            openOutputStream();
+        }
+        if (this.inputStream == null) {
+            openInputStream();
+        }
+
+        byte[] dataInBuffer = flushSerialBuffer();
+        if (dataInBuffer != null && dataInBuffer.length > 0) {
+            logger.warn("eating bytes in the serial buffer input stream before sending command: {}",
+                    getBytesAsString(dataInBuffer));
+        }
+        this.outputStream.write(command, 0, command.length);
+        this.outputStream.flush();
+
+        ByteBuffer buffer = getResponse(timeout);
+        if (buffer != null) {
+            byte[] response = new byte[buffer.limit()];
+            buffer.get(response, 0, response.length);
+            return response;
         } else {
-            throw new NullPointerException("Serial command is null");
+            return null;
         }
     }
 
     @Override
     public synchronized byte[] sendCommand(byte[] command, int timeout, int demark) throws KuraException, IOException {
-        if (command != null) {
-            s_logger.debug("sendMessage() - {}", getBytesAsString(command));
 
-            if (this.m_outputStream == null) {
-                openOutputStream();
-            }
-            if (this.m_inputStream == null) {
-                openInputStream();
-            }
+        if (command == null) {
+            throw new NullPointerException("Serial command must not be null");
```

Comment:
This was old comment about using NPE. I have commented it before we started using exceptions in a different way. Just ignore it 😄.